llvm/lib/MC/MCParser/AsmParser.cpp

```diff
@@ -3021,14 +3021,20 @@ bool AsmParser::parseDirectiveSet(StringRef IDVal, AssignmentKind Kind) {
   return false;
 }
 
-bool AsmParser::parseEscapedString(std::string &Data) {
+bool AsmParser::parseEscapedString(std::string &Data, bool WarnNewline) {
   if (check(getTok().isNot(AsmToken::String), "expected string"))
     return true;
 
   Data = "";
   StringRef Str = getTok().getStringContents();
   for (unsigned i = 0, e = Str.size(); i != e; ++i) {
     if (Str[i] != '\\') {
+      if (Str[i] == '\n' && WarnNewline) {
+        SMLoc NewlineLoc =
+            SMLoc::getFromPointer(getTok().getLoc().getPointer() + i + 1);
```

Comment:
I think you can reduce this to just the following, but need to check:
```suggestion
        SMLoc NewlineLoc = SMLoc::getFromPointer(Str.data() + i);
```

Reply:
Yes, it works, thanks